app/src/main/java/com/example/tanya/godeltechchallengeandroid/domain/interactor/BaseUseCase.kt

```diff
@@ -0,0 +1,18 @@
+package com.example.tanya.godeltechchallengeandroid.domain.interactor
+
+import io.reactivex.Observable
+
+//TODO: Review for possible updates depending on further particular usecases
+abstract class BaseUseCase<Input, Output> {
+
+    fun execute(input: Input): Observable<Output> {
```

Comment:
```suggestion
    fun execute(input: Input? = null): Observable<Output> {
```

After that you can remove the second execute method.